### PR TITLE
set 0045 0046 new

### PR DESCRIPTION
**Pull request by guihao-liang**
code review purpose for #3065;

basically, it checks whether the dependencies exist or not. Otherwise, CMake will complain about the target `""` is not found.

### CODE REVIEW

File: deps/cmake/ExternalProjectOpenSSL.cmake [hunk 1]
```diff
@@ -78,7 +78,7 @@ endif()
 
 target_compile_definitions(openssl INTERFACE HAS_OPENSSL)
 
-add_dependencies(openssl libssla ex_openssl)
+add_dependencies(openssl libssla libcryptoa)
```
Comment (guihao-liang) [inline deps/cmake/ExternalProjectOpenSSL.cmake]: mark to review.

File: cmake/MakeLibrary.cmake [hunk 2]
```diff
@@ -33,98 +35,93 @@ macro(make_target_impl NAME FLAGS REQUIREMENTS IS_LIBRARY SHARED SHARED_ALL_DEFI
...
     target_link_libraries(${NAME} PUBLIC ${REQUIREMENTS})
   endif()
-    
-  # Ensure dependencies are tracked in order to make sure compilation order matters.
-  add_dependencies(${NAME} "${REQUIREMENTS}")
-    
+
+  # Ensure dependencies are tracked in order to make sure compilation order
+  # matters.
+  if (REQUIREMENTS)
```
Comment (guihao-liang) [inline cmake/MakeLibrary.cmake]: mark

File: deps/cmake/ExternalProjectLibXML2.cmake [hunk 3]
```diff
@@ -30,16 +30,15 @@ ExternalProject_Add(ex_libxml2
...
 include_directories(${CMAKE_SOURCE_DIR}/deps/local/include/libxml2)
-add_dependencies(ex_libxml2 ex_libz)
 
 add_library(libxml2a STATIC IMPORTED)
 set_property(TARGET libxml2a PROPERTY IMPORTED_LOCATION ${CMAKE_SOURCE_DIR}/deps/local/lib/libxml2.a)
 
 add_library(libxml2 INTERFACE )
 add_dependencies(libxml2 ex_libxml2)
-target_link_libraries(libxml2 INTERFACE libxml2a z)
+target_link_libraries(libxml2 INTERFACE libxml2a)
```
Comment (guihao-liang) [inline deps/cmake/ExternalProjectLibXML2.cmake]: mark

**Comment by guihao-liang** [comment]
This will be merged once we sign off #3065

**Comment by TobyRoseman** [comment]
@guihao-liang - I'm really not understanding the purpose of this change from your description. Also it looks like a lot of the changes here are just formatting changes. If you're making a large number of formatting changes, it would be helpful to reviewers to do those as a separate commit.

**Comment by guihao-liang** [comment]
> @guihao-liang - I'm really not understanding the purpose of this change from your description. Also it looks like a lot of the changes here are just formatting changes. If you're making a large number of formatting changes, it would be helpful to reviewers to do those as a separate commit.

Yeah, I marked those places that need to be reviewed. Well, the title explains, I set cmake policy 0045 0046 from OLD to NEW, which means it will error out when a target doesn't exist.

**Comment by guihao-liang** [comment]
Passed internal pipeline. Let's merge.

File: cmake/MakeLibrary.cmake [hunk 4]
```diff
@@ -33,98 +35,93 @@ macro(make_target_impl NAME FLAGS REQUIREMENTS IS_LIBRARY SHARED SHARED_ALL_DEFI
...
-#                 REQUIRES libx liby
-#                 MAC_REQUIRES libz libzz
-#                 LINUX_REQUIRES libk libj
-#                 [SHARED] [OUTPUT_NAME xxxx] [SHARED_ALL_DEFINED]
-#                 [OBJECT]
-#                 )
-# Example:
+endmacro()
+
+# This is an external function Usage: make_library(NAME target SOURCES a.cpp
```
Comment (hoytak) [inline cmake/MakeLibrary.cmake:87]: The auto formatting messed up all the tables in the help -- maybe make the change without reformatting the file?  Or figure out how to get it to not mess these up?

File: cmake/MakeLibrary.cmake [hunk 5]
```diff
@@ -33,98 +35,93 @@ macro(make_target_impl NAME FLAGS REQUIREMENTS IS_LIBRARY SHARED SHARED_ALL_DEFI
...
-#   - other libraries built by the the turicreate build system
-#   - builtin libraries
-#   - system libraries
-# MAC_REQUIRES lists all dependent libraries which are included only on Mac.
-# LINUX_REQUIRES lists all dependent libraries which are included only on Linux.
-# SHARED will build a shared library instead of a static library
-# EXTERNAL_VISIBILITY will make the symbols be publicly visible. Default is hidden
-# SHARED_ALL_DEFINED will require shared libraries to have all symbols defined
-# OBJECT will build an object library instead of a static library
+# NAME and SOURCES must be specified. REQUIRES lists all dependent libraries.
```
Comment (hoytak) [inline cmake/MakeLibrary.cmake:95]: Here too.

File: deps/cmake/ExternalProjectOpenSSL.cmake (hunk 1 above)
Comment (hoytak) [reply deps/cmake/ExternalProjectOpenSSL.cmake:81]: ok

File: cmake/MakeLibrary.cmake (hunk 5 above)
Comment (guihao-liang) [reply cmake/MakeLibrary.cmake]: good catch!

File: cmake/MakeLibrary.cmake (hunk 4 above)
Comment (guihao-liang) [reply cmake/MakeLibrary.cmake]: done.

**Pull request merged**